instrumentation/reactor/reactor-netty/reactor-netty-0.9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/reactornetty/v0_9/HttpClientInstrumentation.java

```diff
@@ -74,92 +76,111 @@ public void transform(TypeTransformer transformer) {
       if (DecoratorFunctions.shouldDecorate(callback.getClass())) {
-        callback = new DecoratorFunctions.OnRequestErrorDecorator(callback);
+        return new DecoratorFunctions.OnRequestErrorDecorator(callback);
       }
+      return callback;
     }
   }
 
   @SuppressWarnings("unused")
   public static class OnResponseAdvice {
 
+    @AssignReturned.ToArguments(@ToArgument(0))
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onEnter(
-        @Advice.Argument(value = 0, readOnly = false)
-            BiConsumer<? super HttpClientResponse, ? super Connection> callback,
+    public static BiConsumer<? super HttpClientResponse, ? super Connection> onEnter(
+        @Advice.Argument(0) BiConsumer<? super HttpClientResponse, ? super Connection> callback,
         @Advice.Origin("#m") String methodName) {
       if (DecoratorFunctions.shouldDecorate(callback.getClass())) {
         boolean forceParentContext = methodName.equals("doAfterResponse");
-        callback = new DecoratorFunctions.OnResponseDecorator(callback, forceParentContext);
+        return new DecoratorFunctions.OnResponseDecorator(callback, forceParentContext);
       }
+      return callback;
     }
   }
 
   @SuppressWarnings("unused")
   public static class OnResponseErrorAdvice {
 
+    @AssignReturned.ToArguments(@ToArgument(0))
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onEnter(
-        @Advice.Argument(value = 0, readOnly = false)
-            BiConsumer<? super HttpClientResponse, ? super Throwable> callback) {
+    public static BiConsumer<? super HttpClientResponse, ? super Throwable> onEnter(
+        @Advice.Argument(0) BiConsumer<? super HttpClientResponse, ? super Throwable> callback) {
       if (DecoratorFunctions.shouldDecorate(callback.getClass())) {
-        callback = new DecoratorFunctions.OnResponseErrorDecorator(callback);
+        return new DecoratorFunctions.OnResponseErrorDecorator(callback);
       }
+      return callback;
     }
   }
 
   @SuppressWarnings("unused")
   public static class OnErrorAdvice {
 
+    @AssignReturned.ToArguments({
+      @ToArgument(value = 0, index = 0),
+      @ToArgument(value = 1, index = 1)
+    })
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onEnter(
-        @Advice.Argument(value = 0, readOnly = false)
-            BiConsumer<? super HttpClientRequest, ? super Throwable> requestCallback,
-        @Advice.Argument(value = 1, readOnly = false)
-            BiConsumer<? super HttpClientResponse, ? super Throwable> responseCallback) {
+    public static Object[] onEnter(
+        @Advice.Argument(0)
+            BiConsumer<? super HttpClientRequest, ? super Throwable> originalRequestCallback,
+        @Advice.Argument(1)
+            BiConsumer<? super HttpClientResponse, ? super Throwable> originalResponseCallback) {
+
+      // intermediate variables needed for inlined instrumentation
```

Comment:
was there some side effect that surfaced that caused issues with referencing the originals without making an intermediate?